res/controllers/Pioneer-DDJ-SB3-scripts.js

```diff
@@ -0,0 +1,1791 @@
+            PioneerDDJSB3.valueVuMeter["[Channel2]_enabled"] = 0;
+            PioneerDDJSB3.valueVuMeter["[Channel4]_enabled"] = 0;
+        }
+        if (PioneerDDJSB3.blinkAutodjState === 2) {
+            PioneerDDJSB3.valueVuMeter["[Channel1]_enabled"] = 1;
+            PioneerDDJSB3.valueVuMeter["[Channel3]_enabled"] = 1;
+            PioneerDDJSB3.valueVuMeter["[Channel2]_enabled"] = 1;
+            PioneerDDJSB3.valueVuMeter["[Channel4]_enabled"] = 1;
+        }
+        if (PioneerDDJSB3.blinkAutodjState === 3) {
+            PioneerDDJSB3.valueVuMeter["[Channel1]_enabled"] = 0;
+            PioneerDDJSB3.valueVuMeter["[Channel3]_enabled"] = 0;
+            PioneerDDJSB3.valueVuMeter["[Channel2]_enabled"] = 1;
+            PioneerDDJSB3.valueVuMeter["[Channel4]_enabled"] = 1;
+        }
+    } else {
+        PioneerDDJSB3.valueVuMeter["[Channel1]_enabled"] = 1;
+        PioneerDDJSB3.valueVuMeter["[Channel3]_enabled"] = 1;
+        PioneerDDJSB3.valueVuMeter["[Channel2]_enabled"] = 1;
+        PioneerDDJSB3.valueVuMeter["[Channel4]_enabled"] = 1;
+    }
+};
+
+
+///////////////////////////////////////////////////////////////
+//                        AutoDJ                             //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSB3.autodjSkipNext = function(channel, control, value) {
+    if (value === 0) {
+        return;
+    }
+    if (engine.getValue("[AutoDJ]", "enabled")) {
+        engine.setValue("[AutoDJ]", "skip_next", true);
+    }
+};
+
+PioneerDDJSB3.autodjToggle = function(channel, control, value) {
+    if (value === 0) {
+        return;
+    }
+    if (engine.getValue("[AutoDJ]", "enabled")) {
+        engine.setValue("[AutoDJ]", "enabled", false);
+    } else {
+        engine.setValue("[AutoDJ]", "enabled", true);
+    }
+};
+
+
+///////////////////////////////////////////////////////////////
+//                      CONTROL BINDING                      //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSB3.bindSamplerControlConnections = function(samplerGroup) {
+    engine.makeConnection(samplerGroup, "duration", PioneerDDJSB3.samplerLedsDuration);
+    engine.makeConnection(samplerGroup, "play", PioneerDDJSB3.samplerLedsPlay);
+};
+
+PioneerDDJSB3.bindDeckControlConnections = function(channelGroup, isUnbinding) {
+    var i,
+        index,
+        controlsToFunctions = {
+            "pfl": "PioneerDDJSB3.headphoneCueLed",
+            "keylock": "PioneerDDJSB3.keyLockLed",
+            "loop_enabled": "PioneerDDJSB3.autoLoopLed",
+        };
+
+    controlsToFunctions.slipEnabled = "PioneerDDJSB3.slipLed";
+
+    for (i = 1; i <= 8; i++) {
+        controlsToFunctions["hotcue_" + i + "_enabled"] = "PioneerDDJSB3.hotCueLeds";
+    }
+
+    for (index in PioneerDDJSB3.looprollIntervals) {
+        controlsToFunctions["beatlooproll_" + PioneerDDJSB3.looprollIntervals[index] + "_activate"] = "PioneerDDJSB3.beatlooprollLeds";
```

Comment:
Please do not use strings for referencing a function, use the function reference.

Reply:
These changes are committed.

Reply:
You don't need to write a comment. Just press the "Resolve conversation" button if you followed the request/advice.